Snowflake.Data/Core/HttpUtil.cs

```diff
@@ -109,10 +112,19 @@ private HttpClient RegisterNewHttpClientIfNecessary(HttpClientConfig config)
             return _HttpClients[name];
         }
 
+        private bool IsNetFramework()
+        {
+#if NETFRAMEWORK

```

Comment:
Why conditional compilation? Does it mean we will have two versions of drivers?

Reply:
Yes you have. One for NetFramework, second for Net6.

Reply:
@sfc-gh-igarish I am open to better idea how to solve the issue.

Reply:
There are multiple conditional compilations in this codebase. What's the issue with that?

Reply:
ok. I am checking internally and get back to you ASAP.

Reply:
@myshon  I think it's better to check runtime version with SFEnvironment.ClientEnv.netRuntime which is set here:
https://github.com/snowflakedb/snowflake-connector-net/blob/master/Snowflake.Data/Core/RestParams.cs#L52
#if NETFRAMEWORK is used there as well but it would be easier to just replace the implementation there when we have better solution of checking .net version in the future.
@sfc-gh-igarish Other than that the change looks good to me.

Reply:
Thanks for response. I just fixed it.

Reply:
Thanks. All look good to me now.